File: models/Gpt-3.5-turbo/py/code/sorting_and_searching/Movie_Festival.py

n = int(input())
movies = []

for _ in range(n):
    a, b = map(int, input().split())
    movies.append((b, a))  # store ending time first for sorting

movies.sort()  # sort movies based on ending time

count = 0
end_time = 0

for movie in movies:
    if movie[1] >= end_time:
        count += 1
        end_time = movie[0]

print(count)

File: models/Gpt-3.5-turbo/cpp/code/problems/sorting_and_searching/Movie_Festival.cpp

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int n;
    cin >> n;

    vector<pair<int, int>> movies(n);
    for (int i = 0; i < n; i++) {
        cin >> movies[i].second >> movies[i].first; // we store ending time as second and starting time as first
    }

    sort(movies.begin(), movies.end());

    int answer = 0;
    int latestEndingTime = 0;
    for (int i = 0; i < n; i++) {
        if (movies[i].second >= latestEndingTime) {
            answer++;
            latestEndingTime = movies[i].first;
        }
    }

    cout << answer << endl;

    return 0;
}
